### ISSUE TITLE
Possible bug with comment icons / getCommentText

### ISSUE THREAD

**Issue by maribethb** [google/blockly#7610]
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Description

We have a report from a user that comments do not always deserialize correctly. Check the stack trace below and [this forum post ](https://groups.google.com/g/blockly/c/Rc5WiS8J1kc)for more information.

In `blockToDom`, we call `getCommentText`, which calls `getIcon`. If `getCommentText` returns something falsey, there's no comment, so we move on. If it has a value, then the block must have a comment, so the block must have a comment icon. However, as shown in the forum post, they're seeing `getIcon` return null after `getCommentText` returns with text. That doesn't make sense upon inspection, because `getCommentText` calls `getIcon`! So I really don't know what's happening here.

We did a bit of debugging and it seemed like sometimes the debugger would stop [here](https://github.com/google/blockly/blob/e45998423975ce50e85cbc83ab5aa27799fcb6e0/core/block.ts#L2238) when calling `getIcon`, which implies that the parameter passed is not an `instanceof IconType`, but also inspecting the value of that parameter seemed like it was indeed an instance of the `IconType` class, so I suspect there may be sourcemaps shenanigans happening making debugging trickier.

I've had a look at the [node-red-contrib-blockly repo](https://github.com/bartbutenaers/node-red-contrib-blockly) and I don't see where they're doing any patching or anything suspicious that would imply this is a problem with their code. So I think this is worth looking into on our side because if we do have a problem with deserializing comments, that would be a problem. 

### Reproduction steps

1. Install node-red
2. Install node-red-contrib-blockly
3. Link blockly to the above library so that we can test local changes to blockly
4. Launch both of the above locally somehow
5. Use a blockly node in node-red
6. When you open the blockly node editor, we get a small playground. There's an expand to full screen button. If you click that, that's when the error is triggered. That may play into things here because presumably that button is loading the xml from the smaller playground. But opening the smaller playground to begin with should also trigger an xml load with the saved node data...
7. I have an email with some test data I don't want to post here, but can share if needed to recreate this.

### Stack trace

```shell
Uncaught TypeError: Cannot read properties of undefined (reading 'getBubbleSize') ( xml.ts:194)
    at blockToDom$$module$build$src$core$xml (xml.ts:194:26)
    at blockToDomWithXY$$module$build$src$core$xml (xml.ts:106:19)
    at Object.workspaceToDom$$module$build$src$core$xml [as workspaceToDom] (xml.ts:54:25)
    at Object.open (<anonymous>:819:51)
    at n (red.min.js?v=:19:44660)
    at red.min.js?v=:19:45794
```

**Comment by Apoorvgarg-creator** [comment]
Hello @maribethb Can I take this issue ?

**Comment by maribethb** [comment]
Hi @Apoorvgarg-creator, thanks for your interest! Zoey actually already looked into this and found an issue that points to this being an issue on the node-red-contrib-blockly side rather than a bug on our end, so I'm going to close this issue. Like Beka said we do have other issues open for contribution though!

Thanks for looking into this, Zoey! I can't seem to tag you here. For future reference, details of the problem are in the forum post.

**Comment by alicialics** [comment]
Thank you Maribeth!

To outline the issue:
In node-red script tags are used to instantiate and re-instantiate Blockly when configuration change.
The issue is caused by the old Blockly instance creating a workspace that is being serialized by a newer Blockly instance.  This breaks some assumptions in the code, and cause things like `obj instanceof Klass` to fail.

I proposed to the user that they should only load Blockly once in their system.  If that's difficult to do I am proposing a temporary fix by using the same Blockly instance that created the workspace to serialize it to Xml.  I've created a PR for the temporary fix [here](https://github.com/bartbutenaers/node-red-contrib-blockly/pull/116#issue-1962510296).
